BullsAndCows/BullsAndCowsGame.cs

```diff
@@ -1,20 +1,66 @@
 ﻿using System;
+using System.Linq;
+using System.Text.RegularExpressions;
 
 namespace BullsAndCows
 {
     public class BullsAndCowsGame
     {
         private readonly SecretGenerator secretGenerator;
+        private readonly string secret = string.Empty;
+        private int gameCount;
         public BullsAndCowsGame(SecretGenerator secretGenerator)
         {
             this.secretGenerator = secretGenerator;
+            this.secret = this.secretGenerator.GenerateSecret();
+            gameCount = 1;
         }
 
-        public bool CanContinue => true;
+        public bool CanContinue()
+        {
+            return gameCount < 6 ? true : false;
+        }
+
+        public string Judge(string guess)
+        {
+            var guessWithOutSpace = guess.Replace(" ", string.Empty);
+            gameCount += 1;
+            return this.Answer(this.secret, guessWithOutSpace);
+        }
 
-        public string Guess(string guess)
+        public bool IsInputValid(string guess)
         {
-            throw new NotImplementedException();
+            var rules = new Regex(@"^([0-9]\s){3}[0-9]$");
+            if (rules.IsMatch(guess))
+            {
+                return guess.Split(" ").Distinct().ToArray().Length == 4 ? true : false;
```

Comment:
magic number; the return value of this expression is already bool. Is the "?" required?